skimage/measure/_regionprops.py

```diff
@@ -901,6 +1062,80 @@ def perimeter(image, neighbourhood=4):
     return total_perimeter
 
 
+def perimeter_crofton(image, directions=4):
+    """Calculate total perimeter of all objects in binary image.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        2D image. If image is not binary, all values strictly greater than zero 
+        are considered as the object.
+    directions : 2 or 4, optional
+        Number of directions used to approximate the Crofton perimeter. By 
+        default, 4 is used: it should be more precise than 2. 
```

Comment:
I think it's accurate here rather than precise but I'll let native speakers confirm.

Reply:
https://manoa.hawaii.edu/exploringourfluidearth/physical/world-ocean/map-distortion/practices-science-precision-vs-accuracy